nixos/modules/programs/looking-glass.nix

```diff
@@ -0,0 +1,401 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  isAttrsEmpty = set: (!builtins.any (x: true) (builtins.attrNames set));
+  cleanup = set: attrsets.filterAttrs (n: v: !(isAttrsEmpty v)) (attrsets.filterAttrsRecursive (name: v: v != null) set);
+
+  cfg = config.looking-glass;
+in {
+  options.looking-glass = {
+    enable = mkEnableOption "looking-glass module";
+    package = mkOption {
+      type = types.package;
+      description = "The looking-glass-client package to install";
+      default = pkgs.looking-glass-client;
+    };
+
+    shm = {
+      name = mkOption {
+        type = types.str;
+        description = "Name of shared memory file";
+        default = "looking-glass";
+      };
+      user = mkOption {
+        type = types.str;
+        description = "User of shared memory file";
+        default = "1000";
+      };
+      group = mkOption {
+        type = types.str;
+        description = "Group of shared memory file";
+        default = "libvirtd";
+      };
+      permissions = mkOption {
+        type = types.str;
+        description = "Permission bits for shared memory file";
+        default = "0660";
+      };
+    };
+
+    config = {
```

Comment:
This should be using a freeform module with `pkgs.formats.ini`.

See <https://nixos.org/manual/nixos/stable/index.html#sec-freeform-modules>, <https://nixos.org/manual/nixos/stable/index.html#sec-settings-nix-representable>

Reply:
I think you meant to link the section below but I removed the option spam anyways and kept it simple.